ios/browser/brave_ads/ads_service_impl_ios.mm

```diff
@@ -84,75 +150,96 @@
                              std::move(callback)));
 }
 
-void AdsServiceImplIOS::RunDBTransaction(
-    mojom::DBTransactionInfoPtr mojom_db_transaction,
-    RunDBTransactionCallback callback) {
-  database_.AsyncCall(&brave_ads::Database::RunDBTransaction)
-      .WithArgs(std::move(mojom_db_transaction))
-      .Then(std::move(callback));
+void AdsServiceImplIOS::AddBatAdsObserver(
+    mojo::PendingRemote<bat_ads::mojom::BatAdsObserver>
+    /*bat_ads_observer_pending_remote*/) {
+  NOTIMPLEMENTED();
+}
+
+void AdsServiceImplIOS::GetDiagnostics(GetDiagnosticsCallback callback) {
+  if (!IsRunning()) {
+    return std::move(callback).Run(/*diagnostics*/ std::nullopt);
+  }
+
+  ads_->GetDiagnostics(std::move(callback));
 }
 
 void AdsServiceImplIOS::GetStatementOfAccounts(
     GetStatementOfAccountsCallback callback) {
-  CHECK(IsRunning());
+  if (!IsRunning()) {
+    return std::move(callback).Run(/*statement*/ nullptr);
+  }
```

Comment:
Do we still need these checks, because `AdsImpl` now queues API calls until ads has successfully initialized?

Reply:
Ah... `IsRunning()` checks if `ads_` is valid, so maybe rename `IsRunning` to `IsInitialized`